dask/dataframe/tests/test_arithmetics_reduction.py

```diff
@@ -1161,27 +1165,51 @@ def test_reductions_frame_dtypes():
     ddf_no_timedelta = dd.from_pandas(df_no_timedelta, 3)
 
     assert_eq(df.drop(columns="dt").sum(), ddf.drop(columns="dt").sum())
+    with check_numeric_only_deprecation():
+        expected = df_no_timedelta.drop(columns="dt").mean()
     assert_eq(
-        df_no_timedelta.drop(columns="dt").mean(),
+        expected,
         ddf_no_timedelta.drop(columns="dt").mean(),
     )
 
-    assert_eq(df.prod(), ddf.prod())
-    assert_eq(df.product(), ddf.product())
+    with check_numeric_only_deprecation():
+        expected = df.prod()
+    assert_eq(expected, ddf.prod())
+    with check_numeric_only_deprecation():
+        expected = df.product()
+    assert_eq(expected, ddf.product())
     assert_eq(df.min(), ddf.min())
     assert_eq(df.max(), ddf.max())
     assert_eq(df.count(), ddf.count())
-    assert_eq(df.sem(), ddf.sem())
-    assert_eq(df.sem(ddof=0), ddf.sem(ddof=0))
-
-    assert_eq(df_no_timedelta.std(), ddf_no_timedelta.std())
-    assert_eq(df_no_timedelta.std(skipna=False), ddf_no_timedelta.std(skipna=False))
-    assert_eq(df_no_timedelta.std(ddof=0), ddf_no_timedelta.std(ddof=0))
-    assert_eq(df_no_timedelta.var(), ddf_no_timedelta.var())
-    assert_eq(df_no_timedelta.var(skipna=False), ddf_no_timedelta.var(skipna=False))
-    assert_eq(df_no_timedelta.var(ddof=0), ddf_no_timedelta.var(ddof=0))
+    with check_numeric_only_deprecation():
```

Comment:
This is very busy, why not just wrap the whole block in a single `check_numeric_only_deprecation`?

Reply:
I'm wanting to be targeted about where warnings are being ignored. I thought about doing something like you've suggested and adding an `n_warnings` keyword to count the number of warnings emitted to make sure it's as expected. I've already got the more explicit thing implemented here so I'm inclined to just go with that